### PR TITLE
LESQ-1113: All uk school URN's provided to Avo

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 1970

#### This PR updates the following:
- Updates Avo branch
- Adds two regex functions, to get all UK `schoolUrns` and get `schoolName` from the school Id string

#### Avo events updated:
- curriculumResourcesDownloadedCurriculumDocument
- curriculumResourcesDownloaded
- lessonResourcesDownloaded
- lessonShared

## Issue(s)

Fixes #LESQ-1113

## How to test

1. Go to:

Curriculum downloads:
https://deploy-preview-2879--oak-web-application.netlify.thenational.academy/teachers/programmes/history-primary-ks1/units/traditional-stories-what-do-they-tell-us-about-the-distant-past-1757/lessons/st-patrick-patron-saint-of-ireland/share?preselected=all
https://deploy-preview-2879--oak-web-application.netlify.thenational.academy/teachers/curriculum/previous-downloads?subject=citizenship&keystage=ks4

Teacher downloads / share:
https://deploy-preview-2879--oak-web-application.netlify.thenational.academy/teachers/programmes/chemistry-secondary-ks4-foundation-edexcel/units/atomic-structure-and-the-periodic-table/lessons/atomic-structure-negligible-electron-mass/downloads?preselected=all
https://deploy-preview-2879--oak-web-application.netlify.thenational.academy/teachers/programmes/citizenship-secondary-ks4-l/units/how-well-does-the-media-hold-those-in-power-to-account-63c4/lessons/what-is-the-role-of-the-media-in-a-democracy-cmw32c/share?preselected=all

2. Open console, and click download or share 

3. You should see all download/share events will have have the school URN being sent as a string, Northern Irish schools will be hyphenated

## Screenshots

How it should now look:

Screenshot 2024-10-15 at 12 04 40
Screenshot 2024-10-15 at 12 08 53

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/CurriculumComponents/OakComponentsKitchen/Fieldset/__snapshots__/Fieldset.test.tsx.snap [hunk 1]
```diff
@@ -7,7 +7,7 @@ exports[`Fieldset render 1`] = `
       class="sc-ad0ad6f5-0 sc-3dab0ad0-0 bZaRlP eSpKwY"
     >
```
Comment (k-huggs) [inline src/components/CurriculumComponents/OakComponentsKitchen/Fieldset/__snapshots__/Fieldset.test.tsx.snap:8]: Not sure why these snapshot tests have been failing, has happened on a couple of PRs

File: src/components/TeacherComponents/helpers/downloadAndShareHelpers/getFormattedDetailsForTracking.tsx [hunk 2]
```diff
@@ -31,13 +32,22 @@ export const getSchoolName = (
...
   if (schoolOption === "Selected school") {
-    const schoolName = school.split("-")[1] || "";
+    const schoolName = extractSchool(school) || "";
     return schoolName;
   } else {
     return "";
   }
 };
 
+export function extractUrn(school: string) {
```
[the remark this answers is not in the document]
Comment (k-huggs) [reply src/components/TeacherComponents/helpers/downloadAndShareHelpers/getFormattedDetailsForTracking.tsx]: When people can't read fluent regex

File: src/components/TeacherComponents/helpers/downloadAndShareHelpers/getFormattedDetailsForTracking.test.tsx [hunk 3]
```diff
@@ -60,7 +60,7 @@ describe("getFormattedDetailsForTracking", () => {
     expect(schoolDetailsForTracking).toStrictEqual({
       schoolOption: "Homeschool",
       schoolName: "",
-      schoolUrn: 0,
+      schoolUrn: "",
       selectedResourcesForTracking: [
         "slide deck",
         "starter quiz questions",
```
[the remark this answers is not in the document]
Comment (k-huggs) [reply src/components/TeacherComponents/helpers/downloadAndShareHelpers/getFormattedDetailsForTracking.test.tsx:94]: Added tests ✅

File: src/components/TeacherComponents/helpers/downloadAndShareHelpers/getFormattedDetailsForTracking.tsx [hunk 4]
```diff
@@ -31,13 +32,22 @@ export const getSchoolName = (
...
+  return /^\d{7}|^\d{6}|^\d{3}-\d{4}/.exec(school)?.at(0);
+}
+
+function extractSchool(school: string) {
+  const match = /(?:\d{7}|\d{6}|\d{3}-\d{4})-(.*)/.exec(school);
```
[the remark this answers is not in the document]
Comment (k-huggs) [reply src/components/TeacherComponents/helpers/downloadAndShareHelpers/getFormattedDetailsForTracking.tsx]: combined into one and left a comment on top

File: src/components/TeacherComponents/helpers/downloadAndShareHelpers/getFormattedDetailsForTracking.tsx [hunk 5]
```diff
@@ -31,13 +31,22 @@ export const getSchoolName = (
...
-    const schoolName = school.split("-")[1] || "";
-    return schoolName;
+    const { schoolName } = extractUrnAndSchool(school);
+    return schoolName || "";
   } else {
     return "";
   }
 };
 
+// Regex to extract URN and school name from schoolId string
```
Comment (njblore) [inline src/components/TeacherComponents/helpers/downloadAndShareHelpers/getFormattedDetailsForTracking.tsx:41]:
```suggestion
// Regex to extract URN and school name from schoolId string
// English and Welsh URNs are 6 digits, Scottish URNs are 7 digits and Northern Irish URNs are 6 digits separated by a hyphen
```

**Review by njblore — APPROVED** [review]

**Review by sonalisin — APPROVED** [review]

**Review by jack-skerrett-oak — APPROVED** [review]
School URNs are sent through successfully to posthog, regardless of where the school is in the UK

**Pull request merged**